x-pack/elastic-agent/pkg/capabilities/expr.go

```diff
@@ -0,0 +1,37 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package capabilities
+
+import "strings"
+
+const (
+	wild      = "*"
+	separator = "/"
+)
+
+func matchesExpr(pattern, target string) bool {
+	if pattern == wild {
+		return true
+	}
+
+	patternParts := strings.Split(pattern, separator)
+	targetParts := strings.Split(target, separator)
+
+	if len(patternParts) != len(targetParts) {
+		return false
+	}
```

Comment:
The PR description ends the list of rules with deny on `*`. But we have inputs that will not match. Like `synthetics/*` or `system/metrics`. Do we need to have a 2 final deny rules, one saying `*` and another `*/*`, or shall we allow `*` to capture multiple path elements?

Reply:
* is a special case and is handled on the top and it catches a/b.
* if you use */* ti will match 2 parts constructs